jamaljsr/polar
# feat(payments): add ability to create & pay invoices

### Description

Adds the ability to create and pay invoices from within Polar. I only implemented the bare minimum fields required to create/pay in LND & c-lightning. More fields can easily be added in the future based on specific user requests.

This feature is helpful for those cases when you want to test one side of the transfer from your app. You won't need to use the terminal in Polar for the other side.

### Steps to Test

1. Boot up a network with 2 nodes
2. Create a channel between alice & bob
3. Click on bob, then the Actions tab
4. Click the Create Invoice button, set the amount, then submit the form
5. Click on the Copy & Close button
6. Click on alice, then the Actions tab
7. Click the Pay Invoice button, paste the invoice, then submit the form
8. Confirm the channel updates with the balances adjusted

### Screenshots

![image](https://user-images.githubusercontent.com/1356600/70382534-c2295c00-192b-11ea-997b-6627b62b868b.png)

![image](https://user-images.githubusercontent.com/1356600/70382507-5941e400-192b-11ea-8436-22417ad70464.png)

![image](https://user-images.githubusercontent.com/1356600/70382511-6959c380-192b-11ea-9dff-f423eb8edf7f.png)

![image](https://user-images.githubusercontent.com/1356600/70382520-868e9200-192b-11ea-8895-d6b11e5e93af.png)
